x-pack/plugins/uptime/server/rest_api/overview_filters/get_overview_filters.ts

```diff
@@ -42,17 +42,29 @@ export const createGetOverviewFilters: UMRestApiRouteFactory = (libs: UMServerLi
       }
     }
 
-    return await libs.requests.getFilterBar({
-      uptimeEsClient,
-      dateRangeStart,
-      dateRangeEnd,
-      search: parsedSearch,
-      filterOptions: objectValuesToArrays<string>({
-        locations,
-        ports,
-        schemes,
-        tags,
-      }),
-    });
+    try {
+      return await libs.requests.getFilterBar({
+        uptimeEsClient,
+        dateRangeStart,
+        dateRangeEnd,
+        search: parsedSearch,
+        filterOptions: objectValuesToArrays<string>({
+          locations,
+          ports,
+          schemes,
+          tags,
+        }),
+      });
+    } catch (e) {
+      /**
+       * This particular error is usually indicative of a mapping problem within the user's
+       * indices. It's relevant for the UI because we will be able to provide the user with a
+       * tailored message to help them remediate this problem on their own with minimal effort.
+       */
+      if (e.name === 'ResponseError') {
+        return response.badRequest({ body: e });
+      }
+      throw e;
+    }
   },
 });
```

Comment:
This route has been , infact this whole API has been removed in our filters API, so i wonder if this is the correct appraoch to handle this kind of error.

Reply:
Yes I overlooked that - a dedicated route may be the way to go. The inclination for this approach is, we don't want to slow down the happy path that 99% of users will experience for the sake of this one other case. While it's a common error, it's rare for people to experience it.

Perhaps we should depend on the filters PR and adjust the logic in this PR to fit the error in that one, since the error should still occur with the new fetch procedure.